File: POO JAVA/Aula08/src/aula08/Lutador.java
package aula08;

public class Lutador {

    //Atributos da Classe
    private String nome;
    private String nacionalidade;
    private int idade;
    private float altura;
    private float peso;
    private String categoria;
    private int vitorias;
    private int derrotas;
    private int empates;
    
    //Metodo construtor
    public Lutador(String nome, String nacionalidade, int idade, float altura, float peso, int vitorias, int derrotas, int empates) {
        this.nome = nome;
        this.nacionalidade = nacionalidade;
        this.idade = idade;
        this.altura = altura;
        this.setPeso(peso);
        this.vitorias = vitorias;
        this.derrotas = derrotas;
        this.empates = empates;
    }
    
    //Metodos Acessores e Modificadores
    public String getNome(){    
        return nome;    
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getNacionalidade() {
        return nacionalidade;
    }
    public void setNacionalidade(String nacionalidade) {
        this.nacionalidade = nacionalidade;
    }
    public int getIdade() {
        return idade;
    }
    public void setIdade(int idade) {
        this.idade = idade;
    }
    public float getAltura() {
        return altura;
    }
    public void setAltura(float altura) {
        this.altura = altura;
    }
    public float getPeso() {
        return peso;
    }
    public void setPeso(float peso) {
        this.peso = peso;
        if(this.getPeso() < 50){
            System.out.println("Pesso muito baixo para qualquer categoria.");
        }else if(this.getPeso() >= 50 && this.getPeso() < 70){
            this.categoria = "Peso Pena";
        }else if(this.getPeso() >= 70 && this.getPeso() <= 90){
            this.categoria = "Peso Leve";
        }else if(this.getPeso() >= 90 && this.getPeso() < 110){
            this.categoria = "Peso Medio";
        }else if(this.getPeso() >= 110 && this.getPeso() < 140){
            this.categoria = "Peso Pesado";
        }else{
            System.out.println("Peso muito alto, sem categoia especifica.");
        }
    }
    public String getCategoria() {
        return categoria;
    }
    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }
    public int getVitorias() {
        return vitorias;
    }
    public void setVitorias(int vitorias) {
        this.vitorias = vitorias;
    }
    public int getDerrotas() {
        return derrotas;
    }
    public void setDerrotas(int derrotas) {
        this.derrotas = derrotas;
    }
    public int getEmpates() {
        return empates;
    }
    public void setEmpates(int empates) {    
        this.empates = empates;
    }

    //Metodos da Classe
    public String apresentar() {
        return "\n==== Lutador === {" + "\nnome = " + nome + 
                "\nnacionalidade = " + nacionalidade + "\nidade = " + idade + 
                "\naltura = " + altura + "\npeso = " + peso + 
                "\ncategoria = " + categoria + "\nvitorias = " + vitorias + 
                "\nderrotas = " + derrotas + "\nempates = " + empates + '}';
    
    }
    public void status(){
        System.out.println("\n==========================");
        System.out.println("Nacionalida: "+this.getNacionalidade());
        System.out.println("Nome: "+this.getNome());
        System.out.println("Vitorias: "+this.getVitorias());
        System.out.println("Derrotas: "+this.getDerrotas());
        System.out.println("Empates: "+this.getEmpates());
    }
    public void ganharLuta(){
        this.vitorias++;
    }
    public void perderLuta(){
        this.derrotas++;
    }
    public void empatarLuta(){
        this.empates++;
    }
}

File: POO PHP/Aula08/Lutador.php
<?php

class Lutador {

    //Atributos da Classe
    private $nome;
    private $nacionalidade;
    private $idade;
    private $altura;
    private $peso;
    private $categoria;
    private $vitorias;
    private $derrotas;
    private $empates;
    
    //Metodo Construtor
    public function __construct($nome, $nacionalidade, $idade, $altura, $peso, $vitorias, $derrotas, $empates) {
        $this->nome = $nome;
        $this->nacionalidade = $nacionalidade;
        $this->idade = $idade;
        $this->altura = $altura;
        $this->setPeso($peso);
        $this->vitorias = $vitorias;
        $this->derrotas = $derrotas;
        $this->empates = $empates;
    }

    //Metodos Acessores e Modificadores
    public function getNome() {
        return $this->nome;
    }
    public function getNacionalidade() {
        return $this->nacionalidade;
    }
    public function getIdade() {
        return $this->idade;
    }
    public function getAltura() {
        return $this->altura;
    }
    public function getPeso() {
        return $this->peso;
    }
    public function getCategoria() {
        return $this->categoria;
    }
    public function getVitorias() {
        return $this->vitorias;
    }
    public function getDerrotas() {
        return $this->derrotas;
    }
    public function getEmpates() {
        return $this->empates;
    }
    public function setNome($nome): void {
        $this->nome = $nome;
    }
    public function setNacionalidade($nacionalidade): void {
        $this->nacionalidade = $nacionalidade;
    }
    public function setIdade($idade): void {
        $this->idade = $idade;
    }
    public function setAltura($altura): void {
        $this->altura = $altura;
    }
    public function setPeso($peso): void {
        $this->peso = $peso;
        if($this->peso < 50){
            echo "Peso muito Baixo, lutador sem categoria";
        }elseif ($this->peso >=50 && $this->peso < 70) {
            $this->categoria = "Peso Pena";
        }elseif ($this->peso >= 70 && $this->peso < 90) {
            $this->categoria = "Peso leve";
        }elseif ($this->peso >=90 && $this->peso < 110) {
            $this->categoria = "Peso Medio";
        }elseif ($this->peso >=110 && $this->peso < 130) {
            $this->categoria = "Peso Pesado";
        } else {
            echo "Peso muito Alto, lutador sem categoria";
        }
    }
    public function setCategoria($categoria): void {
        $this->categoria = $categoria;
    }
    public function setVitorias($vitorias): void {
        $this->vitorias = $vitorias;
    }
    public function setDerrotas($derrotas): void {
        $this->derrotas = $derrotas;
    }
    public function setEmpates($empates): void {
        $this->empates = $empates;
    }

    //Metodos da classe
    public function apresentar() {
        echo "<br>*=*=*=*=*=*=*=*=*=*=*=*=*=*";
        echo "<br> = Nome: ".$this->nome;
        echo "<br> = Pais: ".$this->nacionalidade;
        echo "<br> = Idade: ".$this->idade;
        echo "<br> = Altura: ".$this->altura;
        echo "<br> = Peso: ".$this->peso;
        echo "<br> = Categoria: ".$this->categoria;
        echo "<br> = Vitorias: ".$this->vitorias;
        echo "<br> = Derrotas: ".$this->derrotas;
        echo "<br> = Empates: ".$this->empates;
    }
    public function status() {
        echo "<br>==================";
        echo "<br>Nome: ".$this->nome;
        echo "<br>Pais: ".$this->nacionalidade;
        echo "<br>Vitorias: ".$this->vitorias;
        echo "<br>Derrotas: ".$this->derrotas;
        echo "<br>Empates: ".$this->empates;
    }
    public function ganharLuta() {
        $this->vitorias++;
    }
    public function perderLuta() {
        $this->derrotas++;
    }
    public function empatarLuta() {
        $this->empates++;
    }
}
